### PR TITLE
Add DPU support documentation

### PR DESCRIPTION
**Pull request by adrianchiris**
- Add design doc which depicts the current implementation
  in OVN-Kubernetes
- Add DPU section in ovs_offload docs

Note:

Currently only a general overview is provided
in the in-project documents plus links to google doc.

This is a first stage to have some initial documentation
in the project. Later on these docs should be converted
to Markdown and be part of the project.

Signed-off-by: Adrian Chiris <adrianc@nvidia.com>

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: docs/design/dpu_support.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## DPU support
+
+With the emergence of [Data Processing Units](https://blogs.nvidia.com/blog/2020/05/20/whats-a-dpu-data-processing-unit/) (DPU)
```
Comment (Billy99) [inline docs/design/dpu_support.md:3]: First sentence needs to be rewritten, the 'which' doesn't work. Maybe something like:
With the emergence of [Data Processing Units](https://blogs.nvidia.com/blog/2020/05/20/whats-a-dpu-data-processing-unit/) (DPUs), NIC Vendors can now offer greater hardware acceleration capability, flexibility and security.

File: docs/design/dpu_support.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+## DPU support
+
+With the emergence of [Data Processing Units](https://blogs.nvidia.com/blog/2020/05/20/whats-a-dpu-data-processing-unit/) (DPU) 
+which offers greater hardware acceleration capability, flexibility and security. 
+
+It is desirable to leverage DPU in ovn-kubernets to accelerate networking and secure the network control plane.
```
Comment (Billy99) [inline docs/design/dpu_support.md:6]: ovn-kubernets -> OVN-Kubernetes

File: docs/ovs_offload.md [hunk 3]
```diff
@@ -210,3 +210,19 @@ tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
 17:12:56.106705 IP 0.0.0.0.68 > 255.255.255.255.67: BOOTP/DHCP, Request from 62:21:f0:89:40:73, length 30
 ```
 
+## OVS hardware offload DPU support
+
+[Data Processing Units](https://blogs.nvidia.com/blog/2020/05/20/whats-a-dpu-data-processing-unit/) (DPU) combine the advanced capabilities
+of a Smart-NIC (such as Mellanox ConnectX-6DX NIC) with a general purpose embedded CPU and a high-speed memory controller.
+
+Similarly to Smart-NICs, A DPU follows the kernel switchdev model,
```
Comment (Billy99) [inline docs/ovs_offload.md:218]: , A DPU -> , a DPU
model, -> model.

File: docs/ovs_offload.md [hunk 4]
```diff
@@ -210,3 +210,19 @@ tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
...
+In this model for every VF/PF net-device on the host its corresponding representor net-device would exist on
```
Comment (Billy99) [inline docs/ovs_offload.md:219]: Replace with:
In this model, every VF/PF net-device on the host has a corresponding representor net-device existing on the embedded CPU.

File: docs/design/dpu_support.md (hunk 1 above)
Comment (adrianchiris) [reply docs/design/dpu_support.md:3]: Changed.

File: docs/design/dpu_support.md (hunk 2 above)
Comment (adrianchiris) [reply docs/design/dpu_support.md:6]: Changed.

File: docs/ovs_offload.md (hunk 3 above)
Comment (adrianchiris) [reply docs/ovs_offload.md:218]: Changed.

File: docs/ovs_offload.md (hunk 4 above)
Comment (adrianchiris) [reply docs/ovs_offload.md:219]: Changed.

**Comment by adrianchiris** [comment]
still accurate

**Comment by moshe010** [comment]
Looks good to me @Billy99 @zshi-redhat  can you review this PR?

File: docs/design/dpu_support.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+## DPU support
+
+With the emergence of [Data Processing Units](https://blogs.nvidia.com/blog/2020/05/20/whats-a-dpu-data-processing-unit/) (DPUs), 
+NIC vendors can now offer greater hardware acceleration capability, flexibility and security. 
+
+It is desirable to leverage DPU in OVN-kubernets to accelerate networking and secure the network control plane.
```
Comment (Billy99) [inline docs/design/dpu_support.md:6]:
```suggestion
It is desirable to leverage DPU in OVN-kubernetes to accelerate networking and secure the network control plane.
```

File: docs/design/dpu_support.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+A DPU consists of:
+- Industry-standard, high-performance, software-programmable multi-core CPU
+- High-performance network interface
+- Flexible and programmable acceleration engines
+
+Similarly to Smart-NICs, A DPU follows the kernel switchdev model.
```
Comment (Billy99) [inline docs/design/dpu_support.md:13]:
```suggestion
Similarly to Smart-NICs, a DPU follows the kernel switchdev model.
```

**Review by Billy99 — APPROVED** [review]
/lgtm
Minor nits.

File: docs/design/dpu_support.md (hunk 6 above)
Comment (adrianchiris) [reply docs/design/dpu_support.md]: Done

File: docs/design/dpu_support.md (hunk 5 above)
Comment (adrianchiris) [reply docs/design/dpu_support.md]: fixed

**Comment by adrianchiris** [comment]
Addressed @Billy99 comments

**Review by trozet — APPROVED** [review]

**Pull request merged**